tests/backends/mysql/test_introspection.py

```diff
@@ -61,3 +61,115 @@ def test_get_storage_engine(self):
+                            REFERENCES {other_schema_quoted}.{rel_table_quoted}(id)
+                    )
+                    """
+                )
+                relations = connection.introspection.get_relations(
+                    default_cursor, main_table
+                )
+            self.assertEqual(relations, expected_relations)
+        finally:
+            with default_connection.cursor() as default_cursor:
+                default_cursor.execute(f"DROP TABLE {main_table_quoted}")
+            with other_connection.cursor() as other_cursor:
+                other_cursor.execute(f"DROP TABLE {rel_table_quoted}")
+
+
+@skipUnless(connection.vendor == "mysql", "MySQL specific SQL")
+class TestCrossDatabaseConstraints(TestCase):
+    databases = {"default", "other"}
+
+    def test_get_constraints_omits_foreign_schema_constraints(self):
+        """Getting constraints should not include ones to other schemas."""
+        default_connection = connections["default"]
+        other_connection = connections["other"]
+        main_table = "cross_schema_get_constraints_main_table"
+        main_table_quoted = default_connection.ops.quote_name(main_table)
+        other_schema_quoted = other_connection.ops.quote_name(
+            other_connection.settings_dict["NAME"]
+        )
+        rel_table = "cross_schema_get_constraints_rel_table"
+        rel_table_quoted = other_connection.ops.quote_name(rel_table)
+        rel_column = "cross_schema_get_constraints_rel_table_id"
+        rel_column_quoted = other_connection.ops.quote_name(rel_column)
+
+        try:
+            with other_connection.cursor() as other_cursor:
+                # Create table in other schema.
+                other_cursor.execute(f"DROP TABLE IF EXISTS {rel_table_quoted}")
+                other_cursor.execute(
+                    f"""
+                    CREATE TABLE {rel_table_quoted} (
+                        id int(11) NOT NULL auto_increment,
+                        PRIMARY KEY (id)
+                    )
+                    """
+                )
+            with default_connection.cursor() as default_cursor:
+                # Create table in default schema that references other schema.
+                default_cursor.execute(f"DROP TABLE IF EXISTS {main_table_quoted}")
+                default_cursor.execute(
+                    f"""
+                    CREATE TABLE {main_table_quoted} (
+                        id int(11) NOT NULL auto_increment,
+                        {rel_column_quoted} int(11) NOT NULL,
+                        PRIMARY KEY (id),
+                        FOREIGN KEY ({rel_column_quoted})
+                            REFERENCES {other_schema_quoted}.{rel_table_quoted}(id)
+                    )
+                    """
+                )
+                constraints = connection.introspection.get_constraints(
+                    default_cursor, main_table
+                )
+            details = constraints[rel_column]
```

Comment:
The column will still show up as being indexed, even though the FK is empty.

Reply:
Foreign key constraint has a different generated name. I updated tests.